projects/demo-cypress/src/tests/range.cy.ts

```diff
@@ -0,0 +1,311 @@
+import {Component, ElementRef, ViewChild} from '@angular/core';
+import {ComponentFixture} from '@angular/core/testing';
+import {FormControl, ReactiveFormsModule} from '@angular/forms';
+import {TuiRootModule} from '@taiga-ui/core';
+import {TuiKeySteps, TuiRangeComponent, TuiRangeModule} from '@taiga-ui/kit';
+
+describe('TuiRange', () => {
+    let component: TestComponent;
+    let fixture: ComponentFixture<TestComponent>;
+
+    @Component({
+        template: `
+            <tui-root>
+                <tui-range
+                    [formControl]="testValue"
+                    [keySteps]="keySteps"
+                    [max]="max"
+                    [min]="min"
+                    [segments]="segments"
+                    [step]="step"
+                ></tui-range>
+            </tui-root>
+        `,
+    })
+    class TestComponent {
+        @ViewChild(TuiRangeComponent, {static: true})
+        component!: TuiRangeComponent;
+
+        @ViewChild(TuiRangeComponent, {static: true, read: ElementRef})
+        el!: ElementRef<HTMLElement>;
+
+        testValue = new FormControl([3, 5]);
+        max = 11;
+        min = 1;
+        segments = 10;
+        step = 1;
+        keySteps: TuiKeySteps | null = null;
+    }
+
+    beforeEach(() =>
+        cy
+            .mount(TestComponent, {
+                imports: [TuiRootModule, TuiRangeModule, ReactiveFormsModule],
+            })
+            .then(wrapper => {
+                component = wrapper.component;
+                fixture = wrapper.fixture;
+            }),
+    );
+
+    it('The bar is filled from 20% to 40%', () => {
+        expect(getFilledRangeOffset(component).left).to.equal('20%');
+        expect(getFilledRangeOffset(component).right).to.equal('60%');
+    });
+
+    describe('Changing values', () => {
+        describe('Left point', () => {
+            it('Pressing the left arrow decreases the value by one step', () => {
+                cy.get('[automation-id="tui-range__left"]')
+                    .focus()
+                    .type('{leftArrow}')
+                    .then(() => {
+                        expect(component.testValue.value?.[0]).to.equal(2);
+                        expect(component.testValue.value?.[1]).to.equal(5);
+                    });
+            });
+
+            it('Pressing the right arrow increases the value by one step', () => {
+                cy.get('[automation-id="tui-range__left"]')
```

Comment:
**nitpicking:**

Use can use Cypress [aliases](https://docs.cypress.io/api/commands/as) for this repeated selector.

```ts
beforeEach(() => {
    cy.mount(...);

    cy.get('[automation-id="tui-range__left"]').as('leftThumb');
    cy.get('[automation-id="tui-range__right"]').as('rightThumb');
);

// [...]

it('...', () => {
    cy.get('@leftThumb').focus().type('{rightArrow}');
})
```

Reply:
fixed